services/webnn/ort/platform_functions_ort.cc

```diff
@@ -4,28 +4,49 @@
 
 #include "services/webnn/ort/platform_functions_ort.h"
 
+#include "base/base_paths_win.h"
+#include "base/command_line.h"
 #include "base/files/file_path.h"
 #include "base/logging.h"
 #include "base/native_library.h"
 #include "base/path_service.h"
+#include "services/webnn/webnn_switches.h"
 
 namespace webnn::ort {
 
 PlatformFunctions::PlatformFunctions() {
-  // First try to Load onnxruntime.dll from the module folder.
   base::ScopedNativeLibrary ort_library;
-  base::FilePath module_path;
-  if (base::PathService::Get(base::DIR_MODULE, &module_path)) {
-    ort_library = base::ScopedNativeLibrary(base::LoadNativeLibrary(
-        module_path.Append(L"onnxruntime.dll"), nullptr));
-  }
-  if (!ort_library.is_valid()) {
-    ort_library =
-        base::ScopedNativeLibrary(base::LoadSystemLibrary(L"onnxruntime.dll"));
+  base::FilePath ort_library_path;
+
+  // If the switch `kWebNNOrtLibraryPath` is used, try to load onnxruntime.dll
+  // from the specified path firstly.
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kWebNNOrtLibraryPath)) {
+    ort_library_path = base::CommandLine::ForCurrentProcess()
+                           ->GetSwitchValuePath(switches::kWebNNOrtLibraryPath)
+                           .Append(L"onnxruntime.dll");
+    ort_library = base::ScopedNativeLibrary(
+        base::LoadNativeLibrary(ort_library_path, nullptr));
+    if (!ort_library.is_valid()) {
+      LOG(ERROR)
+          << "[WebNN] Failed to load onnxruntime.dll from the specified path: "
+          << ort_library_path;
+    }
   }
+
+  // If the switch `kWebNNOrtLibraryPath` is not used or fail to load
+  // onnxruntime.dll from the specified path, try to load onnxruntime.dll
+  // from the module path.
   if (!ort_library.is_valid()) {
```

Comment:
Consider only loading from module folder when the kWebNNOrtLibraryPath switch is not present. This mode should be mainly used for unit tests.

Reply:
Agree.